### PR TITLE
Added and revised interface doc for GenerateImageClassifierBlackboxSa…

### PR DESCRIPTION
**Pull request by melster1010**
…liency

Took a stab at an intro sentence, also took an educated guess about which implementation was relevant.

### CODE REVIEW

**Review by Purg — CHANGES_REQUESTED** [review]

File: docs/api_docs.rst [hunk 1]
```diff
@@ -136,6 +136,21 @@ A candidate implementation for this interface is the D-RISE [4] algorithm.
...
    :members:
 
+---------------------------------------------
+Blackbox Saliency Image Generation
+---------------------------------------------
+
+Unlike the previous saliency heatmap generation interfaces, this interface uses a blackbox classifier as input along
+with a reference image to generate visual saliency heatmaps .
+
+A candidate implementation for this interface is the D-RISE [4] algorithm.
```
[the remark this answers is not in the document]
Comment (melster1010) [reply docs/api_docs.rst:146]: thanks, I was making an educated guess, guessed wrong :)

**Comment by brianhhu** [comment]
Not tied to additions in this specific PR, but going through the code/documentation for our [two utility masking functions](https://github.com/XAITK/xaitk-saliency/blob/master/xaitk_saliency/utils/masking.py#L13-L194) yesterday, I believe the current stated return output is incorrect. It currently states that both utility functions return a list of `PIL` images, but I believe the correct output is either a numpy array of masked images (batch) or a generator of numpy array masked images (streaming).

We could consider fixing the docstrings as part of this PR.

https://github.com/XAITK/xaitk-saliency/blob/master/xaitk_saliency/utils/masking.py#L68
https://github.com/XAITK/xaitk-saliency/blob/master/xaitk_saliency/utils/masking.py#L159

**Comment by Purg** [comment]
> ... the current stated return output is incorrect. ...

Oops! Good catch.

**Comment by melster1010** [comment]
> Not tied to additions in this specific PR, but going through the code/documentation for our [two utility masking functions](https://github.com/XAITK/xaitk-saliency/blob/master/xaitk_saliency/utils/masking.py#L13-L194) yesterday, I believe the current stated return output is incorrect. It currently states that both utility functions return a list of `PIL` images, but I believe the correct output is either a numpy array of masked images (batch) or a generator of numpy array masked images (streaming).

I changed the docstrings to indicate batch returns "a numpy array of masked images" and streaming returns "a generator of numpy array masked images".

File: docs/api_docs.rst [hunk 2]
```diff
@@ -136,6 +137,22 @@ A candidate implementation for this interface is the D-RISE [4] algorithm.
...
 
+---------------------------------------------
+Blackbox Saliency Image Generation
+---------------------------------------------
+
+Unlike the previous saliency heatmap generation interfaces, this interface uses a blackbox classifier as input along
+with a reference image to generate visual saliency heatmaps .
+
+A candidate implementation for this interface is the PerturbationOcclusion implementation or its sub-implementations
+(RISEStack and/or SlidingWindowStack).
```
Comment (brianhhu) [inline docs/api_docs.rst:148]: Minor nitpick:

```suggestion
(``RISEStack`` and/or ``SlidingWindowStack``).
```

Comment (brianhhu) [inline docs/api_docs.rst:148]: Minor nitpick:

```suggestion
A candidate implementation for this interface is the ``PerturbationOcclusion`` implementation or its sub-implementations
(``RISEStack`` and/or ``SlidingWindowStack``).
```

File: docs/api_docs.rst [hunk 3]
```diff
@@ -53,7 +53,8 @@ We provide two utility functions as baseline implementation to perform this step
 
 * ``xaitk_saliency.utils.masking.occlude_image_batch`` - performs the transformation as a batch operation
 
-* ``xaitk_saliency.utils.masking.occlude_image_streaming`` - performs the transformation in a streaming method with optional parallelization in that streaming
+* ``xaitk_saliency.utils.masking.occlude_image_streaming`` - performs the transformation in a streaming method with
+optional parallelization in that streaming
```
Comment (Purg) [inline docs/api_docs.rst:57]: The second line of text needs to align with the bullet indentation to be considered part of the same block.
```suggestion
* ``xaitk_saliency.utils.masking.occlude_image_streaming`` - performs the transformation in a streaming method with
  optional parallelization in that streaming
```

**Review by Purg — COMMENTED** [review]
Minor issue with bullet formatting.

**Comment by brianhhu** [comment]
LGTM, a rebase will be needed. Thanks!

**Review by Purg — APPROVED** [review]
LGTM

**Pull request merged**